src/lib/search-props.tsx

```diff
@@ -23,7 +24,7 @@ export interface ISearch {
 
 export const getSearchSSRProps =
   (
-    SearchComponent: FunctionComponent<ISearch>,
+    SearchComponent: NextPageWithLayout<ISearch>,
```

Comment:
I think this could be `SearchComponent: NextPage<ISearch>` from `import type { NextPage } from "next";` don't have a dependency on the `getLayout` function inside get `getSearchSSRProps` so can keep it generic.